File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/MyProfileDetailRes.java
package com.team05.petmeeting.domain.user.dto.profile;

import lombok.Builder;

@Builder
public record MyProfileDetailRes(
        Long feedCount,
        Long cheerCount,
        Long feedCommentCount,
        Long animalCommentCount
) {
    public static MyProfileDetailRes of(Long feedCount, Long cheerCount, Long feedCommentCount, Long animalCommentCount) {
        return new MyProfileDetailRes(feedCount, cheerCount, feedCommentCount, animalCommentCount);
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/MyProfileDetailRes.kt
package com.team05.petmeeting.domain.user.dto.profile

data class MyProfileDetailRes(
    val feedCount: Long,
    val cheerCount: Long,
    val feedCommentCount: Long,
    val animalCommentCount: Long
) {
    companion object {
        @JvmStatic
        fun of(
            feedCount: Long,
            cheerCount: Long,
            feedCommentCount: Long,
            animalCommentCount: Long
        ): MyProfileDetailRes {
            return MyProfileDetailRes(feedCount, cheerCount, feedCommentCount, animalCommentCount)
        }
    }
}
